server/service/devices.go

```diff
@@ -335,3 +337,79 @@ func transparencyURL(ctx context.Context, request interface{}, svc fleet.Service
 
 	return transparencyURLResponse{RedirectURL: transparencyURL}, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Get Current Device's MDM Apple Enrollment Profile
+////////////////////////////////////////////////////////////////////////////////
+
+type getDeviceMDMManualEnrollProfileRequest struct {
+	Token string `url:"token"`
+}
+
+func (r *getDeviceMDMManualEnrollProfileRequest) deviceAuthToken() string {
+	return r.Token
+}
+
+type getDeviceMDMManualEnrollProfileResponse struct {
+	// Profile field is used in hijackRender for the response.
+	Profile []byte
+
+	Err error `json:"error,omitempty"`
+}
+
+func (r getDeviceMDMManualEnrollProfileResponse) hijackRender(ctx context.Context, w http.ResponseWriter) {
+	// TODO(mna): does this filename look ok?
+	// make the browser download the content to a file
+	w.Header().Add("Content-Disposition", `attachment; filename="fleet-mdm-enrollment-profile.mobileconfig"`)
+	// explicitly set the content length before the write, so the caller can
+	// detect short writes (if it fails to send the full content properly)
+	w.Header().Set("Content-Length", strconv.FormatInt(int64(len(r.Profile)), 10))
+	// this content type will make macos open the profile with the proper application
+	w.Header().Set("Content-Type", "application/x-apple-aspen-config; charset=urf-8")
+	// prevent detection of content, obey the provided content-type
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+
+	if n, err := w.Write(r.Profile); err != nil {
+		logging.WithExtras(ctx, "err", err, "written", n)
+	}
+}
+
+func (r getDeviceMDMManualEnrollProfileResponse) error() error { return r.Err }
+
+func getDeviceMDMManualEnrollProfileEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	// this call ensures that the authentication was done, no need to actually
+	// use the host
+	if _, ok := hostctx.FromContext(ctx); !ok {
+		err := ctxerr.Wrap(ctx, fleet.NewAuthRequiredError("internal error: missing host from request context"))
+		return getDeviceMDMManualEnrollProfileResponse{Err: err}, nil
+	}
+
+	profile, err := svc.GetDeviceMDMAppleEnrollmentProfile(ctx)
+	if err != nil {
+		return getDeviceMDMManualEnrollProfileResponse{Err: err}, nil
+	}
+	return getDeviceMDMManualEnrollProfileResponse{Profile: profile}, nil
+}
+
+func (svc *Service) GetDeviceMDMAppleEnrollmentProfile(ctx context.Context) ([]byte, error) {
+	// skipauth: service is called by device-authenticated endpoint, adding
+	// SkipAuth call here just to indicate explicitly that no additional
+	// authorization is required.
+	svc.authz.SkipAuthorization(ctx)
+
+	appConfig, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err)
+	}
+
+	mobileConfig, err := generateEnrollmentProfileMobileconfig(
```

Comment:
For future reference, see #9148 and forthcoming research findings regarding profile signing and encryption

Reply:
Thanks for the reference! Does that mean that in the future, the profile will be encrypted and as such won't be able to be directly downloaded and opened in the associated application on the user's device (as there will be a necessary decryption step)? Or is there a recognized MIME type that can open encrypted profiles?

Reply:
This shouldn't prevent a user from directly downloading and opening the profile. We'll have some decisions to make/prioritize in terms of how we want to implement signing and encryption at various points in the flow, but I think we'll probably want to implement signing in the near future at least. Nothing  worry about for this PR though.

With a signed payload, the end-user will see "Verified" in green instead of the scary red warnings in the macOS profile enablement dialog. To get there, the Fleet server will need to be able to sign with a third-party verified certificate. A signing with a self-signed certificate is another possibility but the end-user experience is not ideal since Apple still displays a warning. 

Payloads can be encrypted if the Fleet server knows the public key of an identity certificate in the device's KeyChain. Since Orbit/Fleet Desktop would already be installed, there might be something we could leverage for that certificate here. More generally, I've been looking at how device identity certificates from the SCEP server could be used for encrypting various MDM payloads (which would become available after the initial MDM enrollment meaning that approach could not be used to encrypt the manual enrollment profile). I'm planning to summarize what I've found in comments to the referenced issue.